pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +86,10 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
+
+	return &sourceURL, ensureGitConfigIncludes(gitconfig.Name())
```

Comment:
This is complex enough that we need tests. Opened https://github.com/jhadvig/origin/pull/3 (and https://github.com/openshift/origin/pull/5486 so @csrwng can review) with tests